src/material/chips/chip.scss

```diff
@@ -302,3 +302,10 @@
 .mat-mdc-chip-action:focus .mat-mdc-focus-indicator::before {
   content: '';
 }
+
+// Give the text label a higher z-index than the focus overlay to ensure that the focus overlay
+// does not affect the color of the text. Material spec requires state layer to not interfere with
+// text color.
+.mdc-evolution-chip__text-label {
+  z-index: 1;
```

Comment:
Do we need the same for the other foreground content like avatars, checkmarks and trailing icons?

Reply:
I did not notice any issue with avatars, checkmarks and trailing icons.

Reply:
Looking through this again: can we move the `z-index` into the `.mat-mdc-chip-action-label` selector? It targets the same element, but it's our own selector instead of MDC's.

Reply:
That's a good idea, I changed it to use .mat-mdc-chip-action-label so that it doesn't depend on a class coming from a library.

Reply:
There's already a rule for `.mat-mdc-chip-action-label` a bit further up in the file, you don't need to declare a new one here.

Reply:
good catch, fixed it.